### PR TITLE
moved spi block to where it's needed

### PR DESCRIPTION
**Pull request by slootsky**
without this block moved, there is an error regarding SCK in use

Traceback (most recent call last):
  File "code.py", line 80, in <module>
  File "/lib/adafruit_pyportal/__init__.py", line 152, in __init__
  File "/lib/adafruit_pyportal/network.py", line 95, in __init__
  File "adafruit_portalbase/wifi_coprocessor.py", line 66, in __init__
ValueError: SCK in use

The spi object is passed into the WiFi constructor, but if there was no esp object passed in, then the spi was ignored and attempted to be reconstructed.

Since spi isn't needed in the if esp block, I just moved it to the else block

### CODE REVIEW

**Comment by FoamyGuy** [comment]
Thanks for submitting this fix @slootsky, the change makes sense to me.

I think we also want to use `board.SPI()` instead of `busio.SPI(board.SCK, board.MOSI, board.MISO)` which will help aleviate the issue if the SPI bus has already been initialized before this code is run.

**Comment by slootsky** [comment]
@FoamyGuy  yeah, I saw your patch [  to remove the parameters to board.SPI() ]   after I'd done my code.

I didn't realize that SPI would work with no parameters (I haven't gone into the board code).  I can add that change if you want.

**Comment by FoamyGuy** [comment]
I think the main difference is `board.SPI()` will assume the default pins, but `busio.SPI()` can be passed different ones if you don't want to use the default. 

The other nice thing is `board.SPI()` is forgiving to being called twice, it will not raise an error if you call it twice. Whereas `busio.SPI()` tries to initialize the pin again which raises this error.

I do think we want to make that change. If you change this PR to use `board.SPI()` we can close my PR. I think the way you moved this block into the proper branch of the if statement is a good improvement over mine.

File: adafruit_portalbase/wifi_coprocessor.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def __init__(self, *, status_led=None, esp=None, external_spi=None):
             if external_spi:  # If SPI Object Passed
                 spi = external_spi
             else:  # Else: Make ESP32 connection
-                spi = busio.SPI(board.SCK, board.MOSI, board.MISO)
+                spi = busio.SPI()  # without parameters is forgiving to being called twice
```
Comment (FoamyGuy) [inline adafruit_portalbase/wifi_coprocessor.py:66]: I think this needs to be `board.SPI()` instead of `busio.SPI()`

**Comment by dherrada** [comment]
@slootsky It looks like all you've got to do is run black which should automatically fix the errors we're getting. You can find out how to do that here: https://learn.adafruit.com/improve-your-code-with-pylint/black 

The short version is `pip3 install black` then go to your local repository, then `black --target-version=py35 .`

**Review by FoamyGuy — APPROVED** [review]
These changes look good to me.

I tested them successfully with: 
```
Adafruit CircuitPython 7.0.0-alpha.1-817-g927d5a045 on 2021-04-25; Adafruit PyPortal with samd51j20
```

**Pull request merged**